packages/flutter/test/widgets/overlay_portal_test.dart

```diff
@@ -1365,6 +1365,191 @@ void main() {
     verifyTreeIsClean();
   });
 
+  testWidgets('PortalController can be assigned to another after deactivate', (
+    WidgetTester tester,
+  ) async {
+    final OverlayPortalController controller1 = OverlayPortalController();
+    final OverlayPortalController controller2 = OverlayPortalController();
+
+    late OverlayEntry overlayEntry1, overlayEntry2;
+
+    addTearDown(() {
+      overlayEntry1.dispose();
+      overlayEntry2.dispose();
+    });
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Overlay(
+          initialEntries: <OverlayEntry>[
+            overlayEntry1 = _buildOverlayEntry(
+              (BuildContext context) => OverlayPortal(
+                controller: controller1,
+                overlayChildBuilder: (BuildContext context) => const Placeholder(),
+              ),
+            ),
+            overlayEntry2 = _buildOverlayEntry(
+              (BuildContext context) => OverlayPortal(
+                controller: controller2,
+                overlayChildBuilder: (BuildContext context) => const Placeholder(),
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+
+    await tester.pumpWidget(
```

Comment:
This pumpWidget seems is identical with the one above. Is this intended? Should we update the controllers?

Reply:
No, the controller for both should swap with each other.